index.js

```diff
@@ -1,25 +1,24 @@
-var jsonMask = require('json-mask'),
-  compile = jsonMask.compile,
-  filter = jsonMask.filter;
+const { compile, filter } = require('json-mask');
 
 module.exports = function (opt) {
   opt = opt || {};
 
-  function partialResponse(obj, fields) {
-    if (!fields) return obj;
+  function partialResponse (obj, fields) {
+    if (!fields || this.statusCode !== 200) return obj;
```

Comment:
Let's do this all the way in the beginning, after line 28. Also, all `statusCodes` in the  `[200, 300)` range are valid, so let's include those too.